Structure_Engine/Modify/OrientTowards.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.oM.Base;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Orients the normal of the Bar towards the provided point. The normal is calculated as the vector from the centre of the bar to the provided point.")]
+        [Input("bar", "The Bar to update the orientation of.")]
+        [Input("orientationPoint", "The point to orient the normal of the Bar towards.")]
+        [Output("bar", "Bar with updated orientation. If the calcualtion of the orientation angle fails, the unmodified bar is returned.")]
+        public static Bar OrientTowards(this Bar bar, Point orientationPoint)
+        {
+            Point centre = (bar.StartNode.Position + bar.EndNode.Position) / 2;
```

Comment:
Choice of bar's midpoint as a reference point is arbitrary and will impose a non-perpendicular vector warning in every case where `orientationPoint` is not exactly _above_ the midpoint of the bar. Personally I would prefer not raising the warning at all, regardless of the location of the `orientationPoint`.

Reply:
After discussions offline, get your point. Have change to use the closest point on the finite bar as reference, should limit the warnings to when no line can be drawn from the bar to the point that is orthogonal to the bar